### DISCUSSION TITLE
Überauth Companion Library

### DISCUSSION THREAD

**Discussion by maennchen** [erlef/oidcc#263] [Ideas]
Integration for [`ueberauth`](https://hex.pm/packages/ueberauth)

:pushpin: This issue is here to track interest for the implementation. Upvote if you would like this implemented.

**Comment by paulswartz** [comment]
I started work on this (this use case is my original interest in Oidcc), but external contributions are more than welcome! https://gitlab.com/paulswartz/ueberauth_oidcc/

**Reply by maennchen** [reply]
@paulswartz Awesome!

@yordis Also had some plans around it including redoing some of the ueberauth libraries that are based on OpenID like the google integration if I’m not mistaken. Some coordination would probably help 😊

**Reply by maennchen** [reply]
(PS: Feel free to add the lib to the README here as soon as you consider it stable 😊)

**Reply by yordis** [reply]
@maennchen that will be amazing for sure. Let me know how I could help you.

**Reply by maennchen** [reply]
@paulswartz Is everything ready from your side for a release? I'd do the 3.1.0 as soon as you're ready and have verified our changes.

**Reply by paulswartz** [reply]
Yep, I've been testing with the RC and everything looks good on my end! I'll cut a real 0.1 once 3.1.0 is out.

**Reply by maennchen** [reply]
@paulswartz Released as `v3.1.0`

**Reply by paulswartz** [reply]
@maennchen and I released 0.1: https://hex.pm/packages/ueberauth_oidcc 

Thanks for all the support!

**Reply by maennchen** [reply]
@paulswartz Thanks as well for your work, that is much appreciated :)

**Reply by yordis** [reply]
Probably one of the most important one! I am wondering how many other strategies could be "deprecated" or reuse this one but just swapping the configs?

**Reply by maennchen** [reply]
@yordis True. We could probably "rebase" / wrap a lot of providers around the `oidcc` one while making config easier (predefined config values for issuer, fields etc.)

**Reply by yordis** [reply]
I would be curious about it for sure! @paulswartz if you wish to join Ueberauth org, let me know, we all benefit from collaboration and I definitely needs some help!

**Reply by paulswartz** [reply]
@yordis happy to help out!
